e2e_example/test/common/utils.dart

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+Directory _generatedDir = new Directory(p.join(_toolDir.path, 'generated'));
+Directory _toolDir = new Directory(p.join('.dart_tool', 'build'));
+
+Process _process;
+Stream<String> _stdErrLines;
+Stream<String> _stdOutLines;
+
+/// Runs the build script in this package, and waits for the first build to
+/// complete.
+///
+/// By default, this ensures a brand new build by deleting the `.dart_tool`
```

Comment:
[optional] deleting something by default doesn't match typical user behavior or my expectations (even if it's more commonly what we'd do in an integration test). Consider reversing the default.

Reply:
done